src/config/logging.rs

```diff
@@ -27,14 +28,18 @@ pub fn configure_tracing() -> anyhow::Result<Option<WorkerGuard>> {
         let s = subscriber.with(fmt::layer().with_writer(file_writer));
         s
     };
-
-    let targets_filter = Targets::default().with_default(if cfg!(debug_assertions) {
-        Level::DEBUG
+    let default_level_filter = if cfg!(debug_assertions) {
+        LevelFilter::DEBUG
     } else {
-        Level::INFO
-    });
+        LevelFilter::INFO
+    };
+
+    let env_filter = EnvFilter::builder()
+        .with_default_directive(default_level_filter.into())
+        .with_env_var(LOG_LEVEL_ENV_VAR)
+        .from_env_lossy();
 
-    let subscriber = subscriber.with(targets_filter);
+    let subscriber = subscriber.with(env_filter);
```

Comment:
so i definitely dont understand, but just taking the context of `env_filter` - is the goal of this to modify the logging behavior based on the environment?

Reply:
TBH I don't totally understand the distinction between a `TargetFilter` and an `EnvFilter` either (this is specific to this particular framework and I haven't used it before)

But generally speaking, the `tracing` crate enables something called structured logging within our application.

The big idea behind structured logging is that the system emits logs that have some kind of a structure to them. The "Level" is a part of this structure and will typically fall into one of the following categories:

1. Trace
2. Debug
3. Info
4. Warn
5. Error

Or something similar to them at least.
`Trace` level logs typically contain extremely fine grained detail about what is going on that you wouldn't generally care about unless you had some really specific reason to, `Error` logs indicate that something has gone horribly wrong with your system, and you would pretty much always care about them, and the other levels fall somewhere in between.

You will generally set some log-level threshold to filter out logs that don't rise to at least your level of concern. In our case I set this up to log `Debug` level + logs if you are running a debug build, and `Info` level + logs for release builds, but we also may want a way to adjust these levels without recompiling, and this change utilizes `EnvFilter` to make that possible

The tracing_subscriber docs make it seem like there is more to the two filter types than that, but that is what I was going for here